planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/calculation.cpp

```diff
@@ -326,6 +299,92 @@ double calc_phase_length(
   return std::min(length_with_acceleration, length_with_max_velocity);
 }
 
+std::pair<double, double> calc_min_max_acceleration(
+  const CommonDataPtr & common_data_ptr, const double max_path_velocity,
+  const double prepare_duration)
+{
+  const auto & lc_params = *common_data_ptr->lc_param_ptr;
+  const auto & bpp_params = *common_data_ptr->bpp_param_ptr;
+  const auto current_ego_velocity = common_data_ptr->get_ego_speed();
+
+  const auto min_accel_threshold = std::max(bpp_params.min_acc, lc_params.min_longitudinal_acc);
+  const auto max_accel_threshold = std::min(bpp_params.max_acc, lc_params.max_longitudinal_acc);
+
+  // calculate minimum and maximum acceleration
+  const auto min_acc = calc_minimum_acceleration(
+    lc_params, current_ego_velocity, min_accel_threshold, prepare_duration);
+  const auto max_acc = calc_maximum_acceleration(
+    prepare_duration, current_ego_velocity, max_path_velocity, max_accel_threshold);
+
+  return {min_acc, max_acc};
+}
+
+std::vector<double> calc_acceleration_values(
+  const double min_accel, const double max_accel, const double sampling_num)
+{
+  if (min_accel > max_accel) return {};
+
+  if (max_accel - min_accel < std::numeric_limits<double>::epsilon()) {
+    return {min_accel};
+  }
+
+  const auto resolution = std::abs(max_accel - min_accel) / sampling_num;
+
+  std::vector<double> sampled_values{min_accel};
+  for (double accel = min_accel + resolution;
+       accel < max_accel + std::numeric_limits<double>::epsilon(); accel += resolution) {
+    // check whether if we need to add 0.0
+    if (sampled_values.back() < -eps && accel > eps) {
+      sampled_values.push_back(0.0);
+    }
+
+    sampled_values.push_back(accel);
+  }
+  std::reverse(sampled_values.begin(), sampled_values.end());
+
+  return sampled_values;
+}
+
+std::vector<double> calc_lon_acceleration_samples(
+  const CommonDataPtr & common_data_ptr, const double max_path_velocity,
+  const double prepare_duration)
+{
+  const auto & transient_data = common_data_ptr->transient_data;
+  const auto sampling_num = common_data_ptr->lc_param_ptr->longitudinal_acc_sampling_num;
+
+  const auto [min_accel, max_accel] =
+    calc_min_max_acceleration(common_data_ptr, max_path_velocity, prepare_duration);
+
+  if (max_accel < 0.0) {
+    return calc_acceleration_values(min_accel, max_accel, sampling_num);
+  }
+
+  const auto max_dist_buffer = transient_data.current_dist_buffer.max;
+
+  if (max_dist_buffer > transient_data.dist_to_terminal_end) {
+    return calc_acceleration_values(min_accel, max_accel, sampling_num);
+  }
+
+  if (transient_data.is_ego_stuck) {
+    return calc_acceleration_values(min_accel, max_accel, sampling_num);
```

Comment:
These 3 conditions all return the same `calc_acceleration_values(min_accel, max_accel, sampling_num);` (same as the final return of the function l. 385).
Is there any way to simplify the conditions to make the 2 cases more clear.
- case 1: `return {max_accel}`.
- case 2: `return calc_acceleration_values(min_accel, max_accel, sampling_num);`